src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -6957,6 +6956,20 @@ public void ClearContent(string path)
                     clearStream = false;
                 }
 
+#endif
+                // This block is after we determine if a stream is being cleared because directories
+                // can have data streams that are not child items. They just don't have unnamed data streams.
+                if (Directory.Exists(path)
+#if !UNIX
+                    && !clearStream
+#endif
```

Comment:
The comment is difficult to read. Please reword and add about difference on Windows and Unix.

Reply:
UNIX doesn't have alternate data streams, and so clearStream doesn't even exist as a variable when UNIX is defined.  But on Windows, we need to determine if we're acting on an alternate data stream on a directory before we decide to return out-of-hand.